hub/core/transform/transform.py

```diff
@@ -186,8 +187,12 @@ def run(
         tensors = [target_ds.tensors[t].key for t in tensors]
         group_index = target_ds.group_index
         version_state = target_ds.version_state
-        args = (storage, group_index, tensors, self, version_state, skip_ok)
-        map_inp = zip(slices, repeat(args))
+        if isinstance(storage, MemoryProvider):
+            storages = [storage] * len(slices)
+        else:
+            storages = [storage.copy() for _ in slices]
```

Comment:
creating proper copies of storage to prevent reuse of boto3 resource across workers